pkg/logs/input/docker/launcher.go

```diff
@@ -169,3 +175,45 @@ func (l *Launcher) stopTailer(containerID string) {
 		delete(l.tailers, containerID)
 	}
 }
+
+func (l *Launcher) restartTailer(containerID string) {
+	backoffDuration := 1 * time.Microsecond
+	backoffMax := 60 * time.Microsecond
+	var tailer *Tailer
+	var source *config.LogSource
+
+	for i := 0; ; i++ {
+		if backoffDuration > backoffMax {
+			backoffDuration = backoffMax
+		}
+
+		time.Sleep(backoffDuration)
```

Comment:
You sleep for all restarts here, you should not sleep during the first iteration.